### PR TITLE
Issue #331 Ted/feature password confirmation

### PR DESCRIPTION
**Pull request by tedgusek**
## Description
Implemented a confirm password field on the user signup page along with the corresponding error message if the passwords don't match.  Ticket #331 

Increased MAX output token form 1000 to 1500, as 1000 was not sufficient, this was intended to be a temporary fix, but I propose we leave it for now, until we change the limits for the response to come in under the max output value.

Fixes # (issue)
#331 
## Type of change

- [x] ✨  New feature (non-breaking change which adds functionality)

## How Can this be tested? Testing Plan to review this PR
To test that it stops the process of signing up if the password is not verified, use any sample email when filling out the form, and make sure to use two different password variations in the 'password' and 'confirm password field', you should see a note that requests you to try again because the passwords don't match.  
To ensure that it still creates new members, use a dummy email, and ensure the password is entered the same each time.

- [ ] Where in the app does this change take place?
- [ ] What are the steps to test this change?
- [ ] What is the expected output?

## Checklist:

- [x] I have performed a self-review of my own code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [ ] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [x] I have ensured that my pull request title is descriptive.

### CODE REVIEW

**Review by relandboyle — CHANGES_REQUESTED** [review]
@tedgusek This appears to be working as intended. That's great.

It looks like you're triggering a snack bar notification on clicking the Create an Account button if the password and confirmPassword values don't match. I think I would rather use a Zod validator, so the text fields will provide visual feedback as focus shifts away from a field.

I don't think we need a validator like "minimum 1 character" on confirmPassword, as the only requirement for that field is that it matches the password field.

I'd like to see some more stringent requirements for the password. Something like minimum characters 8, not more than 3 sequential numbers or alphabetical letters, must include uppercase and special chars, etc.

File: server/models/GeminiModel.ts
Comment (kinskiwu) [inline server/models/GeminiModel.ts:1]: If this change is intentional, please address the change in the PR description and the reason. If it was a temp adjustment for local development, please undo the changes.

**Comment by kinskiwu** [comment]
I agree with @relandboyle that a snack bar may not be the appropriate way to show error texts on the form. Colin has implemented a more robust password verification schema in `/validators/validateUserPassword.ts` that can be used in this component. If you haven't already, check out `PasswordUpdatePage.tsx` to see how the confirmed password feature is implemented in the component. 

A couple of minor best practices suggestions:
- Remember check the boxes on the checklist in your PR description
- Consider rebasing your branch since there have been changes on dev since you branched off it.

File: server/models/GeminiModel.ts
Comment (tedgusek) [reply server/models/GeminiModel.ts:1]: This was meant to be a temporary fix, as the response was exceeding the tokens allotment.  However, it might be a good idea to leave it until we can get a better control of the response?
I'm happy to go either way.

**Comment by tedgusek** [comment]
@relandboyle @kinskiwu 
Noted on your comments regarding the code, as well as the best practices.  I'll be sure to keep that in mind moving forward.  I'm going to jump in on this now.

File: server/models/GeminiModel.ts
Comment (relandboyle) [reply server/models/GeminiModel.ts:1]: @tedgusek We can keep the change to the token limit. I see you've already updated the PR description.

File: client/features/auth-pages/SignUpPage.tsx [hunk 1]
```diff
@@ -8,21 +8,21 @@ import { useGithubAuthMutation } from '@hooks/useSignInWithGitHub';
 import { useCreateUserWithEmailAndPasswordMutation } from '@/hooks/useCreateUserWithEmailAndPassword';
 import type { SubmitHandler } from 'react-hook-form';
 import { useForm } from 'react-hook-form';
-import { z } from 'zod';
+// import { z } from 'zod';
```
Comment (kinskiwu) [inline client/features/auth-pages/SignUpPage.tsx]: Pls remove all commented out code

File: client/features/auth-pages/SignUpPage.tsx [hunk 2]
```diff
@@ -38,7 +38,8 @@ const SignUpPage = () => {
     handleSubmit,
     formState: { errors },
   } = useForm<FormData>({
-    resolver: zodResolver(userSignUpSchema),
+    resolver: zodResolver(userPasswordSchema),
+    // resolver: zodResolver(userSignUpSchema),
```
Comment (kinskiwu) [inline client/features/auth-pages/SignUpPage.tsx]: Can you implement the `userEmailSchema` here to validate the email?

File: client/features/auth-pages/SignUpPage.tsx [hunk 3]
```diff
@@ -217,12 +242,12 @@ const SignUpPage = () => {
           </Typography>
         </Box>
       </Box>
-      <SnackBarNotification
+      {/* <SnackBarNotification
```
Comment (kinskiwu) [inline client/features/auth-pages/SignUpPage.tsx]: Any particular reason to move the snackbar inside the box component?

Comment (tedgusek) [reply client/features/auth-pages/SignUpPage.tsx]: No, I just moved it back. I initially wanted to see how it would react when I moved it.

File: client/features/auth-pages/SignUpPage.tsx (hunk 2 above)
Comment (tedgusek) [reply client/features/auth-pages/SignUpPage.tsx]: I just finished up the changes you all pointed out.  Thank you

File: client/features/auth-pages/SignUpPage.tsx [hunk 4]
```diff
@@ -90,7 +111,14 @@ const SignUpPage = () => {
   };
 
   const onSubmit: SubmitHandler<FormData> = async (data) => {
-    const { email, password, name } = data;
+    const { email, password, name, confirmPassword } = data;
+
+    if (password !== confirmPassword) {
```
Comment (kinskiwu) [inline client/features/auth-pages/SignUpPage.tsx]: Since we are using the error texts on the react hook form to display error messages, we don't need the snackBar.

**Review by kinskiwu — APPROVED** [review]
Thanks for the good work! Just a minor change related to the redundant snackBar.

File: client/features/auth-pages/SignUpPage.tsx [hunk 5]
```diff
@@ -8,21 +8,42 @@ import { useGithubAuthMutation } from '@hooks/useSignInWithGitHub';
...
-  email: z.string().email({ message: 'Invalid email address' }),
-  password: z.string().min(1, { message: 'Please enter a password' }),
-  name: z.string().min(1, { message: 'Please enter your name' }),
+const baseSchema = z.object({
+  name: z.string().min(1, { message: `Name is Required` }),
+  email: userEmailSchema.shape.email,
 });
 
-type FormData = z.infer<typeof userSignUpSchema>;
+const combinedSchema = baseSchema.merge(passwordSchema);
```
Comment (relandboyle) [inline client/features/auth-pages/SignUpPage.tsx:38]: @tedgusek You're doing some great work here, and you're close to getting approval.

I have to call out these variable names - combinedSchema, baseSchema, and finalSchema all sound like names that imply ambiguous relationships to other variables. I'd like you to give some thought to whether all three are really necessary, and for all that are necessary please come up with variable names that communicate their purpose or origin more precisely.

Comment (relandboyle) [inline client/features/auth-pages/SignUpPage.tsx]: @tedgusek You're doing some great work here, and you're close to getting approval.

I have to call out these variable names - combinedSchema, baseSchema, and finalSchema all sound like names that imply ambiguous relationships to other variables. I'd like you to give some thought to whether all three are really necessary, and for all that are necessary please come up with variable names that communicate their purpose or origin more precisely.

**Review by relandboyle — CHANGES_REQUESTED** [review]

File: client/features/auth-pages/SignUpPage.tsx (hunk 4 above)
Comment (tedgusek) [reply client/features/auth-pages/SignUpPage.tsx]: Thank you, just made changes and pushed to my branch

File: client/features/auth-pages/SignUpPage.tsx (hunk 5 above)
Comment (tedgusek) [reply client/features/auth-pages/SignUpPage.tsx]: Thank you, I just amended the code, and pushed it up to my branch.

**Comment by relandboyle** [comment]
@tedgusek We're going to close this one because some time has passed and some parts of it have been resolved in other PR's.

Thank you for your work and your perspective on this.
